src/data/polynomial/monic.lean

```diff
@@ -385,8 +385,18 @@ begin
   simp [h.mul_left_ne_zero, hq]
 end
 
-lemma degree_smul_of_non_zero_divisor {S : Type*} [monoid S] [distrib_mul_action S R]
-  (p : polynomial R) (k : S) (h : ∀ (x : R), k • x = 0 → x = 0) :
+lemma monic.is_regular {R : Type*} [ring R] {p : polynomial R} (hp : monic p) : is_regular p :=
+begin
+  split,
+  { intros q r h,
+    rw [←sub_eq_zero, ←hp.mul_right_eq_zero_iff, mul_sub, h, sub_self] },
+  { intros q r h,
+    simp only at h,
+    rw [←sub_eq_zero, ←hp.mul_left_eq_zero_iff, sub_mul, h, sub_self] }
+end
+
+lemma _root_.is_smul_regular.degree_smul {S : Type*} [monoid S] [distrib_mul_action S R]
```

Comment:
If we're going to put this on `is_smul_regular`, it should probably be
```suggestion
lemma _root_.is_smul_regular.polynomial_degree_smul {S : Type*} [monoid S] [distrib_mul_action S R]
```
But I'd be inclined to call it
```suggestion
lemma degree_smul_of_smul_regular {S : Type*} [monoid S] [distrib_mul_action S R]
```